src/SonarScanner.MSBuild.Common/AnalysisProperties/EnvScannerPropertiesProvider.cs

```diff
@@ -35,15 +37,20 @@ public class EnvScannerPropertiesProvider : IAnalysisPropertyProvider
 
         public static bool TryCreateProvider(ILogger logger, out IAnalysisPropertyProvider provider)
         {
+            if (logger == null)
+            {
+                throw new ArgumentNullException(nameof(logger));
+            }
+
             provider = null;
             try
             {
                 provider = new EnvScannerPropertiesProvider(Environment.GetEnvironmentVariable(ENV_VAR_KEY));
                 return true;
             }
-            catch (Exception)
+            catch (Exception ex) when (ex is JsonException || ex is IOException)
             {
-                logger?.LogError(Resources.ERROR_FailedParsePropertiesEnvVar, ENV_VAR_KEY);
+                logger.LogError(Resources.ERROR_FailedParsePropertiesEnvVar, ENV_VAR_KEY, ex.Message);
```

Comment:
this will fail the build, which is ok in this case (otherwise, users will be confused)